### ISSUE TITLE
[JENKINS-46388] GitHub Enterprise, BitBucket Server: Enter key does not submit Server URL when creating a new Pipeline

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/blueocean-plugin#5266]
<p><b>Summary:</b><br/>
When creating a new Pipeline where code is stored on either Bitbucket Server or GitHub Enterprise, if the user tries the Enter key on their keyboard to submit the request, nothing happens.</p>

<p><b>Steps to recreate:</b><br/>
1. Open Blue Ocean and click "New Pipeline"</p>

<p>2. Select either Bitbucket Server, or GitHub Enterprise.</p>

<p>3. Click "Add" to add a new Bitbucket server.</p>

<p>4. Supply a Server Name</p>

<p>5. Supply a Server URL. After entering the Server URL, press the Enter key on your keyboard, as though you're trying to use that to submit the information.</p>

<p>6. Nothing happens. The only way to get the request submitted is by clicking on the Add Server button.</p>

<p> <span class="image-wrap" style=""><a id="39387_thumb" href="https://issues.jenkins.io/secure/attachment/39387/39387_image-2017-08-23-09-11-40-455.png" title="image-2017-08-23-09-11-40-455.png" file-preview-type="image" file-preview-id="39387" file-preview-title="image-2017-08-23-09-11-40-455.png"><img src="https://issues.jenkins.io/secure/thumbnail/39387/_thumb_39387.png" style="border: 0px solid black" role="presentation"/></a></span> </p>

<p><b>Workaround for keyboard navigation geeks like me:</b><br/>
The user can press the Tab key a single time to bring focus to the Add Server button, and submit his request via the Enter key or the space bar.</p>

<p>It could be that I'm an anachronism, and nobody uses the enter key to submit information in this way anymore...</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kshultz.png" title="kshultz's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kshultz">kshultz</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46388" target="_blank">GitHub Enterprise, BitBucket Server: Enter key does not submit Server URL when creating a new Pipeline</a></i></summary>
<i><ul>
<li><b>assignee</b>: <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=nicu">nicu</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Minor
<li><b>component(s)</b>: blueocean-pipeline-editor-plugin
<li><b>label(s)</b>: accessibility, testing-notes-reviewed, usability
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2017-10-16T23:00:47+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 2
<li><b>imported</b>: 2025-12-01
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p><b>Summary:</b>
When creating a new Pipeline where code is stored on either Bitbucket Server or GitHub Enterprise, if the user tries the Enter key on their keyboard to submit the request, nothing happens.</p>

<p><b>Steps to recreate:</b>
1. Open Blue Ocean and click "New Pipeline"</p>

<p>2. Select either Bitbucket Server, or GitHub Enterprise.</p>

<p>3. Click "Add" to add a new Bitbucket server.</p>

<p>4. Supply a Server Name</p>

<p>5. Supply a Server URL. After entering the Server URL, press the Enter key on your keyboard, as though you're trying to use that to submit the information.</p>

<p>6. Nothing happens. The only way to get the request submitted is by clicking on the Add Server button.</p>

<p> <span class="image-wrap" style=""><a id="39387_thumb" href="https://issues.jenkins.io/secure/attachment/39387/39387_image-2017-08-23-09-11-40-455.png" title="image-2017-08-23-09-11-40-455.png" file-preview-type="image" file-preview-id="39387" file-preview-title="image-2017-08-23-09-11-40-455.png"><img src="https://issues.jenkins.io/secure/thumbnail/39387/_thumb_39387.png" style="border: 0px solid black" role="presentation"/></a></span> </p>

<p><b>Workaround for keyboard navigation geeks like me:</b>
The user can press the Tab key a single time to bring focus to the Add Server button, and submit his request via the Enter key or the space bar.</p>

<p>It could be that I'm an anachronism, and nobody uses the enter key to submit information in this way anymore...</p></pre>
</details>
</details>
<details><summary><i>environment</i></summary>

```
Client:<br/>
Fedora Workstation 26, updated at least weekly<br/>
Google Chrome Version 60.0.3112.101 (Official Build) (64-bit)<br/>
Jenkins server:<br/>
Jenkins core 2.74<br/>
blueocean-rest 1.2.0-beta-6<br/>
blueocean-i18n 1.2.0-beta-6<br/>
blueocean-jwt 1.2.0-beta-6<br/>
blueocean-config 1.2.0-beta-6<br/>
blueocean-rest-impl 1.2.0-beta-6<br/>
blueocean-web 1.2.0-beta-6<br/>
blueocean-pipeline-editor 0.3.0-beta-5-SNAPSHOT (private-40e85fc1-kshultz) (1.0-alpha-4)<br/>
blueocean-autofavorite 1.0.0<br/>
blueocean-github-pipeline 1.2.0-beta-6<br/>
blueocean-commons 1.2.0-beta-6<br/>
blueocean-pipeline-api-impl 1.2.0-beta-6<br/>
blueocean-events 1.2.0-beta-6<br/>
blueocean-dashboard 1.2.0-beta-6<br/>
blueocean-pipeline-scm-api 1.2.0-beta-6<br/>
blueocean-personalization 1.2.0-beta-6<br/>
blueocean-git-pipeline 1.2.0-beta-6<br/>
blueocean-display-url 2.1.0<br/>
blueocean-bitbucket-pipeline 1.2.0-beta-6<br/>
blueocean 1.2.0-beta-6
```
</details>
<details><summary><i>1 attachment</i></summary>

- [image-2017-08-23-09-11-40-455.png](https://issues.jenkins.io/secure/attachment/39387/image-2017-08-23-09-11-40-455.png)
  > ![image-2017-08-23-09-11-40-455.png](https://issues.jenkins.io/secure/attachment/39387/image-2017-08-23-09-11-40-455.png)
</details>

**Comment by jenkins-infra-bot** [comment]
- _Remote link associated with this issue:_


  - [Page](https://wiki.jenkins.io/pages/viewpage.action?pageId=98403475)

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>nicu:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46388?focusedId=316330&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-316330">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://github.com/jenkinsci/blueocean-plugin/pull/1466" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/blueocean-plugin/pull/1466</a></p>
</pre>
</details>

</ul>
</details>

<p><a href="https://github.com/jenkinsci/blueocean-plugin/pull/1466" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/blueocean-plugin/pull/1466</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kshultz.png" title="kshultz's avatar" /> kshultz:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46388?focusedId=318810&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-318810">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><b>Testing Notes:</b></p>

<p>It would be worthwhile to add keyboard navigation to some of the automated UI tests. Perhaps code could be added to GithubCreationPage.java, such that there's a way to call <tt><a href="https://github.com/jenkinsci/blueocean-plugin/blob/master/acceptance-tests/src/main/java/io/blueocean/ath/pages/blue/GithubCreationPage.java#L129-L155" class="external-link" target="_blank" rel="nofollow noopener">GithubCreationPage.completeCreationFlow()</a></tt> and specify the use of a <tt>sendKeys(Keys.ENTER)</tt> instead of the existing <tt><a href="https://github.com/jenkinsci/blueocean-plugin/blob/master/acceptance-tests/src/main/java/io/blueocean/ath/pages/blue/GithubCreationPage.java#L105-L107" class="external-link" target="_blank" rel="nofollow noopener">clickCreatePipelineButton()</a></tt>.</p>
</pre>
</details>

</ul>
</details>

<p><b>Testing Notes:</b></p>

<p>It would be worthwhile to add keyboard navigation to some of the automated UI tests. Perhaps code could be added to GithubCreationPage.java, such that there's a way to call <tt><a href="https://github.com/jenkinsci/blueocean-plugin/blob/master/acceptance-tests/src/main/java/io/blueocean/ath/pages/blue/GithubCreationPage.java#L129-L155" class="external-link" target="_blank" rel="nofollow noopener">GithubCreationPage.completeCreationFlow()</a></tt> and specify the use of a <tt>sendKeys(Keys.ENTER)</tt> instead of the existing <tt><a href="https://github.com/jenkinsci/blueocean-plugin/blob/master/acceptance-tests/src/main/java/io/blueocean/ath/pages/blue/GithubCreationPage.java#L105-L107" class="external-link" target="_blank" rel="nofollow noopener">clickCreatePipelineButton()</a></tt>.</p>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `relates_to` from Jira: <a href="https://github.com/jenkinsci/blueocean-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-47811%22">JENKINS-47811</a>]</i>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `epic` from Jira: <a href="https://github.com/jenkinsci/blueocean-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-43955%22">JENKINS-43955</a>]</i>
